homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,215 @@
+                self.hass.data[DOMAIN][SERVERS][server_id] = plex_server
+
+                # ...but do not create a new config entry if it already exists.
+                for entry in self._async_current_entries():
+                    if entry.data[CONF_SERVER_IDENTIFIER] == server_id:
+                        return self.async_abort(  # pylint: disable=lost-exception
+                            reason="already_configured"
+                        )
+
+                url = plex_server.url_in_use
+                token = user_input.get(CONF_TOKEN)
+
+                server_config = {CONF_URL: url}
+                if token:
+                    server_config[CONF_TOKEN] = token
+                if url.startswith("https"):
+                    server_config[CONF_VERIFY_SSL] = user_input.get(
+                        CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL
+                    )
+
+                return self.async_create_entry(  # pylint: disable=lost-exception
+                    title=plex_server.friendly_name,
+                    data={
+                        CONF_SERVER: plex_server.friendly_name,
+                        CONF_SERVER_IDENTIFIER: server_id,
+                        PLEX_SERVER_CONFIG: server_config,
+                    },
+                )
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_TOKEN, default=self.current_login.get(CONF_TOKEN, "")
+                ): str,
+                vol.Optional("manual_setup"): bool,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_manual_setup(self, user_input=None):
+        """Begin manual configuration."""
+        if user_input is None:
+            data_schema = vol.Schema(
+                {
+                    vol.Required(
+                        CONF_HOST, default=self.discovery_info.get(CONF_HOST)
+                    ): str,
+                    vol.Required(
+                        CONF_PORT,
+                        default=int(self.discovery_info.get(CONF_PORT, DEFAULT_PORT)),
+                    ): int,
+                    vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
+                    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+                    vol.Optional(CONF_TOKEN): str,
+                }
+            )
+            return self.async_show_form(step_id="manual_setup", data_schema=data_schema)
+
+        host = user_input.pop(CONF_HOST)
+        port = user_input.pop(CONF_PORT)
+        prefix = "https" if user_input.get(CONF_SSL) else "http"
+        user_input[CONF_URL] = f"{prefix}://{host}:{port}"
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_select_server(self, user_input=None):
+        """Use selected Plex server."""
+        if user_input is None:
+            return await self.async_step_user()
```

Comment:
If input is `None` we should show the form.

Reply:
It needs to accept a list as the argument for the form. Will this latest change work?

Reply:
No. Awaiting this step without user input should show the form with a list of options. When the user selects one of the options, this step will be awaited with that as `user_input`. The user input will never be a list.